app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -483,4 +566,36 @@ class WebViewActivity : AppCompatActivity(), io.homeassistant.companion.android.
             }
         }, 5000)
     }
+
+    private fun setupPanelShortcuts() {
+        waitForConnection()
+        if (isConnected && Build.VERSION.SDK_INT >= 25) {
+            val panels = presenter.getPanels()
+
+            val shortcutManager = getSystemService(ShortcutManager::class.java)
+            val shortcuts = ArrayList<ShortcutInfo>()
+            var count = 0
+            panels.map { panel ->
```

Comment:
Is there a better alternative for this?

Reply:
Syntax might not be perfect because I'm not at my desktop but this would be a proper use of map.
```
val shortcuts = panels
  .filter { !panel.title.isNullOrEmpty() && panel.component_name.contains("lovelace") }
  .take(6)
  .map { 
    ShortcutInfo.Builder(
      this,
      panel.component_name
    )
    .setShortLabel(panel.title!!)
    .setLongLabel(panel.title!!)
    .setIcon(Icon.createWithResource(this, R.drawable.app_icon))
    .setIntent(
      newInstance(this, panel.url_path).apply {
        this.action = Intent.ACTION_VIEW
      }
    )
    .build()
  }
```